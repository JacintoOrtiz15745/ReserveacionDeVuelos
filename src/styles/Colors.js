export const colors = {
  Purple: '#6472ee',
  ligthGray: '#b6b7ba',
  white: '#FFF',
  red: 'red',
  black: 'black',
  
};
