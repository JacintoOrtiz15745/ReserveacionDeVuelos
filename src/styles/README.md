This folder will contain styles if needed.