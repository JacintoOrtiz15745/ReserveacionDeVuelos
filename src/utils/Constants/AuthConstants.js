const GoogleId = '555459120478-5dvb5sj1o0996s86gg8bk69ogm7vdiu4.apps.googleusercontent.com';  
export const AuthConstants = {
  GoogleId,
};
