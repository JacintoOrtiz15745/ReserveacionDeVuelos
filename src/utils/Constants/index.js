
export * from './SignUpConstants';
export * from './FlightsListConstants'
export * from './AuthConstants'
export * from './BookingConstants'