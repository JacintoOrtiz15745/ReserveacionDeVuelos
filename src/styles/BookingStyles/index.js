import {StyleSheet, Dimensions} from 'react-native';
import {colors} from '../Colors';
const {height} = Dimensions.get('window');

export const styles = StyleSheet.create({
  FlightTitle: {
    color: colors.black,
    fontSize: 35,
    fontWeight: 'bold',
    width: 300,
    marginBottom: 80,
  },
  BookingTextInput: {
    height: 70,
    borderBottomColor: colors.black,
    borderBottomWidth: 1,
    fontSize: 20,
    borderBottomColor: colors.ligthGray,
  },
  BookingButton: {
    backgroundColor: colors.Purple,
    padding: 12,
    borderRadius: 10,
    justifyContent: 'center',
  },
  BookingButtonText: {
    textAlign: 'center',
    color: colors.white,
    fontWeight: 'bold',
    fontSize: 18,
  },
  ScreenContainer: {
    margin: 20,
  },
  ContainerInformation: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '50%',
  },
  ContainerFlightInformation: {
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '50%',
  },
  ContainerForm: {
    marginTop: 50,
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '42%',
  },
  dateForm: {
    height: '80%',
    flexDirection: 'column',
    justifyContent: 'flex-start',
  },
  BookingInformation: {
    marginTop: 20,
    marginBottom: 5,
    flexDirection: 'row',
    justifyContent: 'space-between',
    borderBottomWidth: 0.5,
    borderColor: colors.ligthGray,
    height: 70,
  },
  InformationFlightsTitle: {
    fontSize: 25,
    fontWeight: 'bold',
  },
  InformationFlightsSubtitle: {
    color: colors.ligthGray,
    marginTop: 5,
  },
  left: {
    alignSelf: 'flex-start',
  },
  plane: {
    alignSelf: 'center',
  },
  BookingInf: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '55%',
  },
  item: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  FlatList: {
    flexDirection: 'row',
    alignItems: 'center',
    flexDirection: 'row',
    justifyContent: 'center',
    alignContent: 'center',
    width: '50%',
    height: 50,
    alignSelf: 'center',
  },
  title: {
    fontSize: 33,
    fontWeight: 'bold',
  },
  Picker: {
    width: '100%',
    height: 50,
  },
  dateContainer: {
    margin: 20,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    height: 525,
  },
  dateTextContainer: {
    flexDirection: 'column',
  },
  dateText: {
    borderColor: 'black',
    borderWidth: 1,
    marginTop: 10,
    width: 200,
  },
  bold: {
    fontWeight: 'bold',
  },
  doneContainer: {
    padding: 20,
    flexDirection: 'column',
    justifyContent: 'space-between',
    height: '100%',
  },
  doneInfo: {
    paddingTop: height * 0.2,
  },
});
