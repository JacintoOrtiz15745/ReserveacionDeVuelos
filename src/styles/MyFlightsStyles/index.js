import {StyleSheet} from 'react-native';
import {colors} from '../Colors';

export const styles = StyleSheet.create({
  myFlightsTitle: {
    color: colors.Purple,
    fontWeight: 'bold',
    fontSize: 35,
  },
  titleContainer: {
    margin: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  ListContainer: {
    padding: 15,
    height: 520,
  },
  TopContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
  middleContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderBottomWidth: 2,
    borderBottomColor: colors.ligthGray,
    marginTop: 5,
    marginBottom: 10,
  },
  bottomContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 10,
  },
  fatherContainer: {
    marginBottom: 15,
    borderBottomWidth: 2,
  },
  country: {
    fontSize: 30,
    fontWeight: 'bold',
  },
  city: {
    color: colors.greyTitle,
  },
  border: {
    borderBottomColor: 'black',
    borderBottomWidth: 1,
  },
  dateTitle: {
    fontWeight: 'bold',
  },
  passengersTitle: {
    fontWeight: 'bold',
  },
  circleButton: {
    width: 60,
    height: 60,
    borderRadius: 60 / 2,

    alignSelf: 'center',
    backgroundColor: colors.Purple,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
  },
  screenContainer: {
    flexDirection: 'column',
  },
  BookingButton: {
    backgroundColor: colors.Purple,
    padding: 12,
    borderRadius: 10,
    justifyContent: 'center',
  },
  noFlights:{
    fontSize: 25,
    textAlign: 'center',
    color: colors.Purple
  }
});
