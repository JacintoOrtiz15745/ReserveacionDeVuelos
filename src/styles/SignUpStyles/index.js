import {colors} from '../Colors';
import {StyleSheet} from 'react-native';

export const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    color: colors.Purple,
    fontWeight: 'bold',
    marginBottom: 10,
  },
  button: {
    height: 40,
  },
  SignUpContainer: {
    margin: 10,
    padding: 10,
  },
  textInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    marginBottom: 10,
  },
  formText: {
    marginBottom: 10,
    fontSize: 16,
  },
  checkBoxContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'flex-start',
    marginTop: 5,
  },
  btnSignupContainer: {
    marginTop: 15,
  },
  btnSignup: {
    backgroundColor: colors.Purple,
    padding: 12,
    borderRadius: 10,
    marginBottom: 5,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  btnSignupText: {
    textAlign: 'center',
    color: colors.white,
    fontWeight: 'bold',
    fontSize: 18,
  },
  or: {
    textAlign: 'center',
    marginBottom: 10,
    marginTop: 10,
    color: colors.ligthGray,
  },
  LogInText: {
    color: colors.Purple,
  },
  loginTextContainer: {
    textAlign: 'center',
    marginTop: 10,
    flexDirection: 'row',
    justifyContent: 'center',
  },
  googleIcon: {
    width: 20,
    height: 20,
    marginTop: 3,
  },
  characters: {
    marginBottom: 20,
  },
  checkboxFather: {
    marginBottom: 15,
  },
  errorMessage: {
    color: colors.red,
  },
  signin: {
    margin: 10,
    padding: 10,
  },
  signinContainer:{
    marginBottom: 20,
    marginTop: 20,
    flexDirection: 'column',
    justifyContent: 'flex-start',
    height: 450
  }
});
