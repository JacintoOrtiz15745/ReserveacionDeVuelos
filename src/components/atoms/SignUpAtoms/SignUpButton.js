import React from 'react';
import {Text, TouchableOpacity} from 'react-native';
import {colors} from '../../../styles/Colors';
import {styles} from '../../../styles/SignUpStyles';
import {SignUpConstants} from '../../../utils/Constants';
import {connect} from 'react-redux';
import {signupFunction} from '../../../../redux/actions/AuthActions';

const SignUpButton = ({
  isButtonEnabled,
  formValues,
  signupFunction,
}) => {
  return (
    <TouchableOpacity
      testID="SignUpBtn"
      style={[
        styles.btnSignup,
        {backgroundColor: isButtonEnabled ? colors.Purple : colors.ligthGray},
      ]}
      onPress={() => {
        isButtonEnabled
          ? signupFunction(formValues)
          : alert('Invalid Credentials');
      }}>
      <Text testID="btnTitle" style={styles.btnSignupText}>
        {SignUpConstants.SignUpButton}
      </Text>
    </TouchableOpacity>
  );
};
const mapDispatchToProps = {
  signupFunction,
};
export default connect(null, mapDispatchToProps)(SignUpButton);
