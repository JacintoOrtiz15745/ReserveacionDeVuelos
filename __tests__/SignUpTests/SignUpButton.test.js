import React from 'react';
import SignUpButton from '../../src/components/atoms/SignUpAtoms/SignUpButton';
import {shallow} from 'enzyme';
import {SignUpConstants} from '../../src/utils/Constants';
import {colors} from '../../src/styles/Colors';
import {connect} from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';

describe('SignUpButton component renders as expected', () => {
  const mockStore = configureMockStore([]);
  let store;
  store = mockStore({});
  const ConnectedComponent = connect()(SignUpButton);
  const wrapper = shallow(<ConnectedComponent store={store} />);
  const component = wrapper.dive().dive();
  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('button exists', () => {
    const btn = component.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    expect(btn.length).toBe(1);
  });
  test('button contains a title', () => {
    const btnTitle = component.findWhere(
      (node) => node.prop('testID') === 'btnTitle',
    );
    expect(btnTitle.length).toBe(1);
  });
  test('button title matches with our constant', () => {
    const btnTitle = component.findWhere(
      (node) => node.prop('testID') === 'btnTitle',
    );
    expect(btnTitle.props().children).toBe(SignUpConstants.SignUpButton);
  });
});

describe('Funcionality when is not enabled', () => {
  const mockStore = configureMockStore([]);
  let store;
  store = mockStore({});
  const mockFormValues = {
    name: '',
    email: '',
    password: '',
    termsCheckbox: false,
    updatesCheckbox: false,
    validName: false,
    validEmail: false,
    validPassword: false,
    validatedName: false,
    validatedEmail: false,
    validatedPassword: false,
  };
  const signUpSubmitMock = jest.fn();
  const ConnectedComponent = connect()(SignUpButton);
  const wrapper = shallow(
    <ConnectedComponent
      store={store}
      isButtonEnabled={false}
      formValues={mockFormValues}
      signUpSubmit={signUpSubmitMock}
    />,
  );
  const component = wrapper.dive().dive();

  test('buttton color matches with our color constant', () => {
    const btn = component.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    expect(btn.props().style[1].backgroundColor).toEqual(colors.ligthGray);
  });
  test('submit function is not called', () => {
    const btn = component.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    btn.props().onPress();
    component.update();
    expect(signUpSubmitMock).not.toHaveBeenCalled();
  });
});

describe('Funcionality when is enabled', () => {
  const mockStore = configureMockStore([thunk]);
  let store;
  store = mockStore({
    dispatch: jest.fn(),
  });
  const mockFormValues = {
    name: 'juan',
    email: 'juan@mail.com',
    password: 'mm23mm222',
    termsCheckbox: true,
    updatesCheckbox: true,
    validName: true,
    validEmail: true,
    validPassword: true,
    validatedName: true,
    validatedEmail: true,
    validatedPassword: true,
  };

  const ConnectedComponent = connect()(SignUpButton);
  const wrapper = shallow(
    <ConnectedComponent
      store={store}
      isButtonEnabled={true}
      formValues={mockFormValues}
    />,
  );
  const component = wrapper.dive().dive();

  test('buttton color matches with our color constant', () => {
    const btn = component.findWhere(
      (node) => node.prop('testID') === 'SignUpBtn',
    );
    expect(btn.props().style[1].backgroundColor).toEqual(colors.Purple);
  });
});
