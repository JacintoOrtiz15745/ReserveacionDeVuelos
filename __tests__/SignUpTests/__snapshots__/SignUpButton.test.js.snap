// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpButton component renders as expected renders correctly 1`] = `
<Connect(SignUpButton)
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
